frontend/src/pages/Projects.tsx

```diff
@@ -11,19 +12,36 @@ import { API_URL } from '../utils/credentials.ts'
 export default function Projects() {
   const [projectData, setProjectData] = useState<ProjectDataType | null>(null)
   const [defaultProjects, setDefaultProjects] = useState<ProjectDataType | null>(null)
+  const [initialQuery, setInitialQuery] = useState<string>('')
 
   useEffect(() => {
     document.title = 'OWASP Projects'
+    const queryParams = new URLSearchParams(window.location.search)
+    const urlQuery = queryParams.get('q')
+    if (urlQuery) {
+      setInitialQuery(urlQuery)
+    }
+
     const fetchApiData = async () => {
       try {
-        const response = await fetch(`${API_URL}/owasp/search/project`)
-        const data = await response.json()
+        let response
+        if (urlQuery) {
+          response = await axios.get(`${API_URL}/owasp/search/project`, {
+            params: { q: urlQuery },
+          })
+        } else {
+          response = await fetch(`${API_URL}/owasp/search/project`)
+          response = await response.json()
+        }
```

Comment:
Could you narrow it down to `params` difference only based on `urlQuery`. 

BTW, why do you use both `axios` and `fetch`?

Reply:
i have used axios in both one api is for without query and another one is with query